src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -298,6 +371,34 @@ private static void ChangeColor(bool foreground, ConsoleColor color)
         /// <summary>Cache of the format strings for foreground/background and ConsoleColor.</summary>
         private static readonly string[,] s_fgbgAndColorStrings = new string[2, 16]; // 2 == fg vs bg, 16 == ConsoleColor values
 
+        public static bool TryGetSpecialConsoleKey(char[] givenChars, int startIndex, int endIndex, out ConsoleKey key, out int keyLength)
+        {
+            key = (ConsoleKey)0;
+            keyLength = 0;
+            int unprocessedCharCount = endIndex - startIndex + 1;
+
+            if (unprocessedCharCount < TerminalKeyInfo.Instance.MinKeyLength)
```

Comment:
Actually, I am not sure if we will hit this. with vmin = 1 and vtime=0, non-canonical mode and stdin read being atomic in nature, I would expect a key stroke to either be completely captured or completely ignored. The only time I think this case is possible is if we had more than 255 bytes in the buffer at the time we read it. 
However, if this case is possible, how will we differentiate between something getting added to stdin after our read and before our poll and the case where something just got missed from our read.